crates/ra_analysis/src/imp.rs

```diff
@@ -305,6 +305,51 @@ impl AnalysisImpl {
             .collect()
     }
 
+    pub fn resolve_callable(&self, file_id: FileId, offset: TextUnit, token: &JobToken)
+        -> Option<(FnDescriptor, Option<usize>)> {
+
+        use ra_syntax::SyntaxKind;
+        use ra_syntax::ast::{ArgListOwner, Expr};
+        use ra_syntax::text_utils::{contains_offset_nonstrict};
+
+        let root = self.root(file_id);
+        let file = root.syntax(file_id);
+        let syntax = file.syntax();
+
+        // Find the calling expression and it's nameref
+        let call_expr: ast::CallExpr = find_node_at_offset(syntax, offset)?;
+
+        let name_ref = match call_expr.expr()? {
+            Expr::PathExpr(path_expr) => {
+                path_expr.path()?.segment()?.name_ref()?
+            },
+            _ => return None
+        };
+
+        // Resolve the function's NameRef
+        let file_symbols = self.index_resolve(name_ref, token);
+        for (_, fs) in file_symbols {
+            if fs.kind == SyntaxKind::FN_DEF {
+                if let Some(fn_def) = find_node_at_offset(syntax, fs.node_range.start()) {
+                    let descriptor = FnDescriptor::new(fn_def);
+
+                    // If we have a calling expression let's find which argument we are on
+                    let mut current_parameter = None;
+                    if let Some(arg_list) = call_expr.arg_list() {
+                        // NOTE: This call sites like: "(3, " are interpreted as having argument so completion
+                        // will not occur until a value is entered after the comma.
+                        current_parameter = arg_list.args()
+                            .position(|arg| contains_offset_nonstrict(arg.syntax().range(), offset));
+                    }
```

Comment:
This is definitely wrong. I'm not sure if `arg_list()` is the right solution here. Should I manually scan to see how many commas we are in? I also don't think this takes `self` into account but I haven't written that test yet.

Reply:
Hm so this is wrong for cases where there's whitespace/comments around the parameters, and the cursor is on a whitespace? I don't think there's a really neat solution here, just writing some code looking at the offsets of commans, plus a battery of tests, seems like the best we can do. There's `flip_comma` intention which also deals with comma-separated lists. I don't think we can share any code between the two, but it makes sense to check just in case. 

In IntelliJ, we use [this](https://github.com/intellij-rust/intellij-rust/blob/b1ce136d7332a2b031688e00443625ce076ed904/src/main/kotlin/org/rust/ide/hints/RsParameterInfoHandler.kt#L97-L110) for parameter info and have some generic comma list utils [here](https://github.com/intellij-rust/intellij-rust/blob/2524d8531cb3b68621dd2fd633f96f801a5eff89/src/main/kotlin/org/rust/ide/formatter/impl/utils.kt#L108-L135).

Reply:
Ok I have some slightly better logic for this that I will push soon.